A6.md

```diff
@@ -0,0 +1,398 @@
+
+If all instances of a hedged RPC fail, there are no additional retry attempts. Essentially, hedging can be seen as retrying the original RPC before a failure is even received. 
+
+If server pushback that specifies not to retry is received in response to a hedged request, no further hedged requests should be issued for the call.
+
+Hedged requests should be sent to distinct backends, if possible.
+
+#### Throttling Retry Attempts and Hedged RPCs
+
+gRPC prevents server overload due to retries and hedged RPCs by disabling these policies when the client’s ratio of failures to successes passes a certain threshold. The throttling is done per server name. Retry throttling may be configured as follows:
+
+```
+'retry_throttling': {
+  'max_tokens': 10,
+  'token_ratio': 0.1
+}
+```
+
+Throttling may only be specified per server name, rather than per method or per service.
+
+For each server name, gRPC maintains a `token_count` which is initially set to `max_tokens`. Every outgoing RPC (regardless of service or method invoked) will effect `token_count` as follows:
+* Every failed RPC will decrement the `token_count` by 1. 
+* Every successful RPC will increment the `token_count` by `token_ratio`. 
+
+If `token_count` is less than or equal to the threshold, defined to be `(max_tokens / 2)`, then RPCs will not be retried until `token_count` rises over the threshold.
+
+Throttling also applies to hedged RPCs. The first outgoing RPC will always be sent, but subsequent hedged RPCs will only be sent if `token_count` is greater than the threshold.
+
+Neither retry attempts or hedged RPCs block when `token_count` is less than or equal to the threshold. Retry attempts are canceled and the failure returned to the client application. The hedged request is cancelled, and if there are no other already-sent hedged RPCs the failure is returned to the client application.
+
+The only RPCs that are counted as failures for the throttling policy are RPCs that fail with a status code that qualifies as a retryable or non-fatal status code (see [here](#retryable-status-codes) and [here](#hedging-policy)), or that receive a pushback response indicating not to retry. This avoids conflating server failure with responses to malformed requests (such as the `INVALID_ARGUMENT` status code).
+
+#### Pushback
+Servers may explicitly pushback by setting metadata in their response to the client. The pushback can either tell the client to retry after a given delay or to not retry at all. If the client has already exhausted its `max_retry_attempts`, the call will not be retried even if the server says to retry after a given delay.
+
+A new metadata key, `"x-grpc-retry-pushback-ms"`, will be added to support server pushback. If the value for pushback is set to -1, then it will be seen as the server asking the client not to retry at all.
```

Comment:
Before I got to "Throttling Configuration" below, it wasn't really clear to me what's the scope of pushback on the server side. I think it's worth mentioning here since this section is supposed to describe it in details, not the examples one below.